pkg/kv/kvserver/store_create_replica.go

```diff
@@ -25,10 +25,25 @@ import (
 
 var errRetry = errors.New("retry: orphaned replica")
 
-// getOrCreateReplica returns a replica for the given RangeID, creating an
-// uninitialized replica if necessary. The caller must not hold the store's
-// lock. The returned replica has Replica.raftMu locked and it is the caller's
-// responsibility to unlock it.
+// getOrCreateReplica returns a replica with the given replicaID for the given
+// rangeID, or roachpb.RaftGroupDeletedError if this replicaID has been deleted.
+// If the replica is not in memory, but also is not deleted, it creates and
+// returns an uninitialized replica. The returned replica's Replica.raftMu is
+// locked, and the caller is responsible for unlocking it.
+//
+// If getOrCreateReplica returns a replica, the guarantee is that the following
```

Comment:
This makes it sound like these invariants are specific to having been returned from this method, but I think they'rer eally just invariants of holding raftMu: any mutation to the raft state or state machine needs to go through raftMu. This includes destroying the replica (unfortunately we have to logically lump in the RangeTombstone with the raft state even though it's resident on the state machine engine).

Reply:
Maybe these things can be discussed on `raftMu` and we can refer to it here?

Reply:
Agree. The "Locking notes" in `store.go` cover this. I made a note that these invariants are derived from `raftMu` invariants and friends. Still think it's useful to duplicate this info a bit, because it didn't immediately dawn on me even though I knew there were locking notes.

I think the "feature" that this method adds is the fact that all the checks appear atomic with the `raftMu` locked. If we did all the checks, but with not `raftMu` locked all the way, I'm not sure we would end up with the same guarantees. I tried to communicate that in a paragraph above, WDYT?